hopr/hopr-lib/src/lib.rs

```diff
@@ -1203,6 +1203,11 @@ impl Hopr {
         Ok(self.transport_api.ticket_statistics().await?)
     }
 
+    // DB ============
+    pub fn hopr_db(&self) -> HoprDb {
```

Comment:
Since you decided to expose `HoprDb`, I wouldn't expose it as a whole, but only as `HoprDbResolverOperations`.

So you could do smth like:

````rust
pub fn peer_resolver(&self) -> &impl HoprDbResolverOperations {
  &self.db
}
````

Reply:
An alternative approach is not to expose `HoprDb` at all and create a method in the `Hopr` object.

````rust
pub fn fulfill_peer_or_address(&self, poa: PeerOrAddress) -> PeerOrAddress {
   poa.fulfill(&self.db)
}
````

Reply:
Your call. But I'm slightly in favor of the second approach.